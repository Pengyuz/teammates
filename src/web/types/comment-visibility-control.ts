/**
 * The visibility controls.
 */
export enum CommentVisibilityControl {

  /**
   * Show response visibility control.
   */
  SHOW_COMMENT = 'SHOW_COMMENT',

  /**
   * Show giver name visibility control.
   */
  SHOW_GIVER_NAME = 'SHOW_GIVER_NAME',
}
