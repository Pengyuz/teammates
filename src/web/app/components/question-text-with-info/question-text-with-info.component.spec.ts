import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { By } from '@angular/platform-browser';
import { RouterTestingModule } from '@angular/router/testing';
import { FeedbackQuestionType } from '../../../types/api-output';
import {
  QuestionAdditionalInfoModule,
} from '../question-types/question-additional-info/question-additional-info.module';
import { QuestionTextWithInfoComponent } from './question-text-with-info.component';

describe('QuestionTextWithInfoComponent', () => {
  let component: QuestionTextWithInfoComponent;
  let fixture: ComponentFixture<QuestionTextWithInfoComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [QuestionTextWithInfoComponent],
      imports: [
        RouterTestingModule,
        QuestionAdditionalInfoModule,
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(QuestionTextWithInfoComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should not show control link and question details for TEXT questions', () => {
    const questionDetailControlLink: any = fixture.debugElement.query(By.css('a'));
    const explicitQuestionDetail: any = fixture.debugElement.query(By.css('div'));
    expect(questionDetailControlLink).toBeNull();
    expect(explicitQuestionDetail).toBeNull();
  });

  it('should show control link when question type is not TEXT', () => {
    component.questionDetails.questionType = FeedbackQuestionType.MCQ;
    component.questionDetails.questionText = 'MCQ question details';
    fixture.detectChanges();
    const questionDetailControlLink: any = fixture.debugElement.query(By.css('a'));
    const explicitQuestionDetail: any = fixture.debugElement.query(By.css('div'));
    expect(questionDetailControlLink).not.toBeNull();
    expect(explicitQuestionDetail).toBeNull();
    expect(fixture).toMatchSnapshot();
  });

  it('should show question detail when click control link', () => {
    component.questionDetails.questionType = FeedbackQuestionType.MCQ;
    component.questionDetails.questionText = 'MCQ question details';
    fixture.detectChanges();
    const questionDetailControlLink: any = fixture.debugElement.query(By.css('a'));
    questionDetailControlLink.nativeElement.click();
    fixture.detectChanges();
    const explicitQuestionDetail: any = fixture.debugElement.query(By.css('div'));
    const mcqQuestionDetail: any = fixture.debugElement.query(By.css('tm-mcq-question-additional-info'));
    expect(questionDetailControlLink).not.toBeNull();
    expect(explicitQuestionDetail).not.toBeNull();
    expect(mcqQuestionDetail).not.toBeNull();
    expect(fixture).toMatchSnapshot();
  });
});
