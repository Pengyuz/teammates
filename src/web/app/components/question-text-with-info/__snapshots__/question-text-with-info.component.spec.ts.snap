// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionTextWithInfoComponent should show control link when question type is not TEXT 1`] = `
<tm-question-text-with-info
  FeedbackQuestionType={[Function Object]}
  additionalInfoIsExpanded="false"
  questionDetails={[Function Object]}
  questionNumber="0"
>
  Question 0: MCQ question details
  <a
    href="/"
    ng-reflect-query-params-handling="merge"
    queryparamshandling="merge"
  >
     [more]
  
  </a>
</tm-question-text-with-info>
`;

exports[`QuestionTextWithInfoComponent should show question detail when click control link 1`] = `
<tm-question-text-with-info
  FeedbackQuestionType={[Function Object]}
  additionalInfoIsExpanded={[Function Boolean]}
  questionDetails={[Function Object]}
  questionNumber="0"
>
  Question 0: MCQ question details
  <a
    href="/"
    ng-reflect-query-params-handling="merge"
    queryparamshandling="merge"
  >
     [less]
  
  </a><div>
    
    
    
    <tm-mcq-question-additional-info
      ng-reflect-question-details="[object Object]"
    >
      Multiple-choice (single answer) question options:
  
      
      <div>
         The options for this question is automatically generated from the list of all %s in this course.
  
      </div>
      
    </tm-mcq-question-additional-info>
    
    
    
    
    
    
  </div>
</tm-question-text-with-info>
`;
